src/System.Text.Json/System/Text/Json/JsonReader.cs

```diff
@@ -97,7 +97,7 @@ public bool Read()
             ref byte next = ref Unsafe.Add(ref bytes, skip);
             length -= skip;
 
-            int step = GetNextCharAscii(ref next, length, out char ch);
+            int step = GetNextCharAscii(this, ref next, length, out char ch);
             if (step == 0) return false;
```

Comment:
This one is interesting.
`JsonReader` is a mutable struct and `GetNextCharAscii` is a formally mutating method, even though it does not mutate.
No-mixing rules assume here that the callee may wrap the `ch` in a span, directly or indirectly and "poison" `this`, with the method can do, since this is writable.

I could solve this by making `GetNextCharAscii` a readonly method, but we do not support methods that are individually readonly.

I could not make it a 'in this' extension either since it reads private members and I did not want to add another class, although that could be the least disruptive solution if we had a lot of callers.

So I made it a static method that takes the struct as `in` parameter.